### PR TITLE
Changed Object to use Cow<str> instead of String

### PR DESCRIPTION
**Pull request by emoon**
- [x] Tests created for any new feature or regression tests for bugfixes.
- [x] `cargo test` succeeds
- [x] `rustup run nightly cargo clippy` succeeds (prints some warnings)
- [ ] `cargo fmt -- --write-mode=diff` succeeds (reports: Unrecognized option: 'write-mode')

### CODE REVIEW

**Comment by epage** [comment]
> This is a implementation for #186 Problem with this is that it breaks the current API.

Breaking the API is fine at this stage.

So I don't forget this when releasing a new version, do you mind using a [conventional changelog format](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular), including a `BREAKING CHANGE` section?

**Comment by epage** [comment]
Also, if you are ok with relicensing your work to Apache + MIT could you post to https://github.com/cobalt-org/liquid-rust/issues/7

I need to hunt down the last couple of people so I don't have to track this anymore :(

File: src/value/values.rs [hunk 1]
```diff
@@ -57,7 +57,7 @@ impl Value {
             }
             Value::Object(ref x) => {
                 let arr: Vec<String> = x.iter()
-                    .map(|(k, v)| k.clone() + ": " + &v.to_string())
+                    .map(|(k, v)| k.as_ref().to_owned() + ": " + &v.to_string().to_owned())
```
[the remark this answers is not in the document]
Comment (emoon) [reply src/value/values.rs]: Will fix!

File: src/interpreter/context.rs [hunk 2]
```diff
@@ -168,7 +168,7 @@ impl Context {
 
     /// Sets a value in the global context.
     pub fn set_global_val(&mut self, name: &str, val: Value) -> Option<Value> {
-        self.globals.insert(name.to_owned(), val)
+        self.globals.insert(name.to_owned().into(), val)
```
[the remark this answers is not in the document]
Comment (emoon) [reply src/interpreter/context.rs]: Good idea. Will fix

File: src/interpreter/context.rs [hunk 3]
```diff
@@ -181,7 +181,7 @@ impl Context {
     /// this should never happen in a well-formed program.
     pub fn set_val(&mut self, name: &str, val: Value) -> Option<Value> {
         match self.stack.last_mut() {
-            Some(frame) => frame.insert(name.to_owned(), val),
+            Some(frame) => frame.insert(name.to_owned().into(), val),
```
[the remark this answers is not in the document]
Comment (emoon) [reply src/interpreter/context.rs]: Will do!

**Comment by emoon** [comment]
> Also, if you are ok with relicensing your work to Apache + MIT could you post to #7

Added!

**Comment by epage** [comment]
Looks good.  Thanks for pushing individual commits so I could more easily review the changes.

Now that it all looks good, if you don't mind squashing your commits and adjusting your commit message as mentioned, this is good for merging!

**Comment by emoon** [comment]
Cool. I can squash the commits before the final merge. I wonder if the updated message is ok? I'm a bit confused by the page and having that `perf(pencil)` tag? Maybe it should be in the top commit message or?

**Comment by epage** [comment]
>  I wonder if the updated message is ok? I'm a bit confused by the page and having that perf(pencil) tag? Maybe it should be in the top commit message or?

I'm not seeing an updated commit message.  Did you remember to force-push?

My quick run down:
- It would make sense to have this all in one commit
- That commit would have a message like:
- Conventional commit messages are a way for me to easily spot what needs to go into the changelog.  The tag in particular tells me whether it needs to be in release notes (fix, feat) or not (chore, style).

```
perf(value): Reduce allocations with `Cow`

Change `value::Object`'s key to a `Cow` to avoid allocations when a `&'static str` is used.

BREAKING CHANGE: When inserting items into a `value::Object`, the value needs to be turned into a `Cow<str>`, usually with `.into()`. This carries forward to inserting values into a `Context`.
```

Does that clarify things?

**Comment by emoon** [comment]
Yeah. Updated and pushed now.

**Pull request merged**